src/tokens/TokenDetails.tsx

```diff
@@ -0,0 +1,325 @@
+          tokenId={tokenId}
+          style={styles.balance}
+          testID="TokenDetails/Balance"
+        />
+        {!token.isStableCoin && <PriceInfo token={token} />}
+        <Actions token={token} />
+        <Text style={styles.yourBalance}>{t('tokenDetails.yourBalance')}</Text>
+        <TokenBalanceItem token={token} containerStyle={styles.tokenBalanceItem} />
+        {token.infoUrl && (
+          <LearnMore
+            tokenName={token.name}
+            infoUrl={token.infoUrl}
+            analyticsProps={getTokenAnalyticsProps(token)}
+          />
+        )}
+      </ScrollView>
+    </SafeAreaView>
+  )
+}
+
+TokenDetailsScreen.navigationOptions = {
+  ...headerWithBackButton,
+}
+
+function PriceInfo({ token }: { token: TokenBalance }) {
+  const { t } = useTranslation()
+  if (!token.priceUsd) {
+    return (
+      <View style={styles.priceInfo}>
+        <Text style={styles.priceInfoUnavailable}>{t('tokenDetails.priceUnavailable')}</Text>
+      </View>
+    )
+  }
+
+  if (!token.historicalPricesUsd || !isHistoricalPriceUpdated(token)) {
+    return null
+  }
+
+  return (
+    <View style={styles.priceInfo}>
+      <PercentageIndicator
+        comparedValue={token.historicalPricesUsd.lastDay.price}
+        currentValue={token.priceUsd!}
+        percentageTextStyle={styles.priceInfoText}
+        UpIcon={DataUp}
+        DownIcon={DataDown}
+        NoChangeIcon={DataUp}
+        suffixText={t('tokenDetails.priceDeltaSuffix') ?? undefined}
+        suffixTextStyle={typeScale.bodySmall}
+        testID="TokenDetails/PriceDelta"
+      />
+    </View>
+  )
+}
+
+function Actions({ token }: { token: TokenBalance }) {
+  const { t } = useTranslation()
+  const sendableTokens = useSendableTokens()
+  const swappableTokens = useSwappableTokens()
+  const cashInTokens = useCashInTokens()
+  const cashOutTokens = useCashOutTokens()
+  const isSwapEnabled = useSelector(isAppSwapsEnabledSelector)
+
+  const actions = [
+    {
+      name: TokenDetailsActionName.Send,
+      text: t('tokenDetails.actions.send'),
+      iconComponent: QuickActionsSend,
+      onPress: () => {
+        navigate(Screens.Send, { defaultTokenOverride: token.address! })
+      },
+      visible: !!sendableTokens.find((tokenInfo) => tokenInfo.tokenId === token.tokenId),
+    },
+    {
+      name: TokenDetailsActionName.Swap,
+      text: t('tokenDetails.actions.swap'),
+      iconComponent: QuickActionsSwap,
+      onPress: () => {
+        navigate(Screens.SwapScreenWithBack, { fromTokenId: token.tokenId })
+      },
+      visible:
+        isSwapEnabled &&
+        !!swappableTokens.find((tokenInfo) => tokenInfo.tokenId === token.tokenId) &&
+        token.balance.gt(TOKEN_MIN_AMOUNT),
+    },
+    {
+      name: TokenDetailsActionName.Add,
+      text: t('tokenDetails.actions.add'),
+      iconComponent: QuickActionsAdd,
+      onPress: () => {
+        navigate(Screens.FiatExchangeAmount, {
+          currency: token.symbol as CiCoCurrency,
```

Comment:
this type cast seems to rely on the internal logic of `useCashInTokens`, which could change and cause a bug here...

Reply:
same with the following one

Reply:
added an explicit condition on the navigate function